### PR TITLE
create serve script for credential-service and use base64 for environment variable

### PR DESCRIPTION
**Pull request by josephperrott**
See individual commits.

### CODE REVIEW

File: apps/credential-service/lib/server.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ assert(process.env.GITHUB_APP_ID, 'GITHUB_APP_ID is not defined in the environme
 const GITHUB_APP_ID = process.env.GITHUB_APP_ID;
 /** The PEM key of the Github app used to generating tokens. */
 assert(process.env.GITHUB_APP_PEM, 'GITHUB_APP_PEM is not defined in the environment');
-const GITHUB_APP_PEM = process.env.GITHUB_APP_PEM;
+const GITHUB_APP_PEM = Buffer.from(process.env.GITHUB_APP_PEM, 'base64').toString('utf-8');
```
Comment (devversion) [inline apps/credential-service/lib/server.ts:38]: Any reason to encode the secret as base64 in the environment? Please clarify in code/or here, and also rename the environment variable to e.g. `GITHUB_APP_PRIVATE_KEY_BASE64`

File: apps/credential-service/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+if [ -z $(which gcloud) ]; then
+  echo "###################################################################"
+  echo "# Failed:                                                         #"
+  echo "#  gcloud must be installed in order to deploy this service        #"
```
Comment (devversion) [inline apps/credential-service/deploy.sh:4]:
```suggestion
  echo "#  gcloud must be installed in order to deploy this service       #"
```

File: apps/credential-service/lib/server.ts (hunk 1 above)
Comment (josephperrott) [reply apps/credential-service/lib/server.ts:38]: Done

File: apps/credential-service/lib/server.ts [hunk 3]
```diff
@@ -33,9 +33,12 @@ const PORT = 8080;
...
 assert(process.env.GITHUB_APP_ID, 'GITHUB_APP_ID is not defined in the environment');
 const GITHUB_APP_ID = process.env.GITHUB_APP_ID;
-/** The PEM key of the Github app used to generating tokens. */
-assert(process.env.GITHUB_APP_PEM, 'GITHUB_APP_PEM is not defined in the environment');
-const GITHUB_APP_PEM = process.env.GITHUB_APP_PEM;
+/**
+ * The PEM key of the Github app used to generating tokens, the value is stored in the environment
+ * variable in Base64 to make it easier to manage it being a multiline string.
+ */
+assert(process.env.GITHUB_APP_PEM_BASE64, 'GITHUB_APP_PEM is not defined in the environment');
```
Comment (devversion) [inline apps/credential-service/lib/server.ts:40]:
```suggestion
assert(process.env.GITHUB_APP_PEM_BASE64, '`GITHUB_APP_PEM_BASE64` is not defined in the environment');
```

**Review by devversion — APPROVED** [review]

**Pull request closed without merging**